src/code/z_play_hireso.c

```diff
@@ -1,27 +1,1349 @@
+        }
+        if ((i == (this->cursorPageRow + cursorPage)) && (this->scrollAmount == 0)) {
+            barRectLeft = 56;
+            photoOffset = 8;
+            photoRectDs = 200.0f;
+            photoWidth = 64;
+        } else {
+            barRectLeft = 48;
+            photoOffset = 0;
+            photoRectDs = 150.0f;
+            photoWidth = 48;
+        }
+        BombersNotebook_TextureRectangle(&gfx, (57 - photoOffset) * 4, (rectTop - photoOffset) * 4,
+                                         (57 - photoOffset + photoWidth) * 4, (rectTop - photoOffset + photoWidth) * 4,
+                                         0, 0, 0, 1024.0f / (photoRectDs / 100.0f), 1024.0f / (photoRectDs / 100.0f));
+
+        if ((i == (this->cursorPageRow + cursorPage)) && (this->scrollAmount == 0)) {
+            gDPSetPrimColor(gfx++, 0, 0, sBarColorR, sBarColorG, sBarColorB, 255);
+        } else {
+            gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, 60);
+        }
+        gDPLoadTextureBlock(gfx++, gBombersNotebookBarTex, G_IM_FMT_I, G_IM_SIZ_8b, 8, 4, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                            G_TX_NOMIRROR | G_TX_WRAP, 3, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+        BombersNotebook_TextureRectangle(&gfx, (barRectLeft + 57) * 4, (rectTop + 22) * 4,
+                                         (barRectLeft - photoOffset + 527) * 4, (rectTop + 26) * 4, 0, 0, 0, 0x400,
+                                         0x400);
+        gDPLoadTextureBlock_4b(gfx++, gBombersNotebookCircleTex, G_IM_FMT_I, 16, 16, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                               G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+        BombersNotebook_TextureRectangle(&gfx, (barRectLeft - photoOffset + 527) * 4, (rectTop + 16) * 4,
+                                         (barRectLeft - photoOffset + 543) * 4, (rectTop + 32) * 4, 0, 0, 0, 0x400,
+                                         0x400);
+
+        if (CHECK_WEEKEVENTREG(gBombersNotebookWeekEventFlags[i])) {
+            gDPPipeSync(gfx++);
+            gDPSetRenderMode(gfx++, G_RM_OPA_SURF, G_RM_OPA_SURF2);
+            gDPSetPrimColor(gfx++, 0, 0, 255, 255, 255, 255);
+
+            BombersNotebook_DrawEntries(&gfx, i, rectTop);
+
+            gDPPipeSync(gfx++);
+            gDPSetRenderMode(gfx++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+        }
+    }
+    gDPPipeSync(gfx++);
+    *gfxP = gfx;
+}
+
+TexturePtr sBombersNotebookDigitTextures[] = {
+    gBombersNotebookDigit0Tex, gBombersNotebookDigit1Tex, gBombersNotebookDigit2Tex, gBombersNotebookDigit3Tex,
+    gBombersNotebookDigit4Tex, gBombersNotebookDigit5Tex, gBombersNotebookDigit6Tex, gBombersNotebookDigit7Tex,
+    gBombersNotebookDigit8Tex, gBombersNotebookDigit9Tex,
+};
+u8 sBombersNotebookDigitRectLeftOffsets[] = { 13, 8, 12, 13, 14, 13, 12, 12, 13, 12 };
+s16 sBombersNotebookDayTexRectLeftOffsets[] = { 16, 12, 9 };
+
+#define CURRENT_DAY_MIN_1 ((CURRENT_DAY == 0) ? 1 : CURRENT_DAY)
```

Comment:
I would have expected `CURRENT_DAY_MIN_1` to be `(CURRENT_DAY - 1)` (current day minus 1), what's the naming intention here?

Reply:
I was thinking `MIN` for `MINIMUM`. TBH, I had a difficult time trying to name it and couldn't think of anything better.

Reply:
Yeah this is tricky. Maybe `CURRENT_DAY_CLAMP_MIN_1`? But it's not a true clamp.

Reply:
Personally, I like `CURRENT_DAY_CLAMP_MIN_1` the best. I agree the intention is to avoid UB reads (though it is used directly without the -1 a couple of times), but since it only rounds up 0 I think the name should reflect that. (It is also curious that there is a `CURRENT_DAY_CLAMP_MIN_1 >= 3` check instead of just a `== 3`.)